### ISSUE TITLE
[Sug]Share Exp/Gold

### ISSUE THREAD

**Issue by moepi2k** [suriyun-production/mmorpg-kit-docs#1914]
it would be nice to have shared exp/gold based on damage output optional.
healers feels punishing since they never receive exp/gold in groupe because they do not do any damage

https://gyazo.com/8565a1064b606666f7ac5985df68d0db

**Comment by xafgun** [comment]
Or all people in range can get exp/gold, for example member at town can't get exp or gold at that way.

**Comment by insthync** [comment]
I don't get it, enable this one then EXP will be shared, still not enough?

![Screenshot_405](https://user-images.githubusercontent.com/2697550/226505886-9c8e3e1f-ae53-4ca1-8f09-dc5ad575a74a.png)

**Comment by moepi2k** [comment]
Yes thats good, the problem is exp share amount is depends on damage output. When u have 2 players in group 1 healer and 1 tank, and the healer do zero damage because he just healing the tank, then he never will get exp

**Comment by insthync** [comment]
While sharing EXP, it will be shared to everyone equally isn't it ?

**Comment by moepi2k** [comment]
no it shared by dmg output
https://gyazo.com/c2783873930c66da72e93932f98fd650

**Comment by insthync** [comment]
https://youtu.be/lBLkcxGL5Y4

**Comment by insthync** [comment]
> no it shared by dmg output https://gyazo.com/c2783873930c66da72e93932f98fd650
> 
> in the video i just see there seems something wrong with the amount. normaly the chicken give 50 exp. when both player attack it one player get almost half amount of exp and the other the full 50. also the gold taken from chicken is doubled.

For this issues, check the latest commit in Core repo

**Comment by moepi2k** [comment]
> https://youtu.be/lBLkcxGL5Y4

Ok thats weird. Thus video was recored before the core change u mention?

I will check when at home. Could be thats an issue in my project then, just strange that other ppls in discord had same issue

**Comment by insthync** [comment]
Call them to show their issues here.

**Comment by moepi2k** [comment]
> No

So then that means you had same issue?

**Comment by insthync** [comment]
Only having problems while not sharing EXP.

**Comment by insthync** [comment]
And it doesn't mean the EXP sharing feature not working properly.

**Comment by moepi2k** [comment]
ok so i tested it again. the thing with sharing exp is fine when enabled in party. the problem is when the share exp is not enabled or play solo. the exp given from monster is inconsistent. like when u set 50 exp for a monster and kill it alone or even in group the exp given from the monster is random calculated depends on damage output. but it never give you the full amount of exp what you set in monster scriptable

**Comment by moepi2k** [comment]
yea your right, wrong topic and the share exp is working fine. but i tested it with your fix already and it never give the amount of exp what you set in scriptable. 

also a weird behavior is this -> https://gyazo.com/001d7af8aad2daf9bcf28642a5f63303
when you have a weapon with high amount of damage (in this case a one hit) the exp will calcuate in negative. -> check exp bar in the top bar.

dont know if this is maybe related.

**Comment by insthync** [comment]
LAST COMMIT

**Comment by moepi2k** [comment]
```csharp
if (damage > CurrentHp)
                        damage = CurrentHp;
```
yes i added this and video was recorded with that fix

**Comment by insthync** [comment]
No, LAST COMMIT

**Comment by moepi2k** [comment]
lol ok, last commit appears just right now, wastn able to see it before xD

sorry then
